packages/waas/src/auth.ts

```diff
@@ -171,10 +185,91 @@ export class SequenceWaaS {
   ) {
     this.config = defaultArgsOrFail(config)
     this.waas = new SequenceWaaSBase({ network: 1, ...config }, this.store, this.cryptoBackend, this.secureStoreBackend)
-    this.client = new WaasAuthenticator(this.config.rpcServer, this.fetch.bind(this))
+    this.client = new WaasAuthenticator(this.config.rpcServer, this._fetch)
+
     this.deviceName = new StoreObj(this.store, '@0xsequence.waas.auth.deviceName', undefined)
   }
 
+  _fetch = (input: RequestInfo, init?: RequestInit): Promise<Response> => {
+    if (this.signatureVerificationFailed) {
+      throw new Error('Signature verification failed')
+    }
+
+    if (this.cryptoBackend && this.config.disableHttpSignatureCheck !== true && init?.headers) {
+      const headers: { [key: string]: any } = {}
+
+      headers['Accept-Signature'] = 'sig=();alg="rsa-v1_5-sha256"'
+
+      init!.headers = { ...init!.headers, ...headers }
```

Comment:
Was there a reason you didn't just do:

```
init.headers['Accept-Signature'] = 'sig=();alg="rsa-v1_5-sha256"'
```

instead?

Reply:
no reason, i'll change 👍